### PR TITLE
feat: #38 convert local setup to regen ledger

### PR DESCRIPTION
**Pull request by haveanicedavid**
Closes: #38 

- convert local dev setup to use regen-ledger
- update README and makefiles 
- adds `make install-local-ledger` command to clone regen ledger into a sub-folder and use that binary for local dev

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,34 +26,34 @@
...
-## Install Cosmos-SDK
+## Install Regen Ledger
 
-`git clone https://github.com/cosmos/cosmos-sdk` and **within that directory**:
+`git clone https://github.com/regen-network/regen-ledger/` and **within that directory**:
 
-1. `git checkout v0.46.11`
-2. `make build`
+1. `make build` - build ledger locally
+2. `mv build/regen build/regenlocal` - rename to `regenlocal` to avoid conflicts with the `regen` binary
```
Comment (clevinson) [inline README.md:34]: Why is it necessary to have a separate local version of the binary?  If someone wants to use the existing `regen` binary in their path, shouldn't that be just fine?

I think the best approach is to either:
1. Install the latest verison of Regen binary (if not already installed), and ensure it is accessible in the $PATH
2. Use the already installed version of regen binary on your system

Any local configuration we want for testing should be doable by setting up a local network with its own `$REGEN_HOME`.  As long as the env variable of `$REGEN_HOME` is set in the context of testing the groups-ui, we should be able to have locally configured network for local testing that works fine.

Probably it makes sense to not store that home directory in the user's home dir (e.g. `~/.regenlocal`, but rather put it relative to this repo, and also include it in the .gitignore.

Adding something like `export REGEN_HOME=$(pwd)/.regen` to the makefile  should do the trick.

Comment (haveanicedavid) [reply README.md:34]: > Why is it necessary to have a separate local version of the binary? If someone wants to use the existing regen binary in their path, shouldn't that be just fine?

I wasn't sure - tbh I'm not super familiar with the differences in how regen / cosmos operate locally vs when pointed at the network, and haven't worked a ton with makefiles / binaries in general, and this was mainly done for my own peace of mind 😄 

Everything you've said here makes sense - appreciate your feedback @clevinson , will update 👍

Comment (clevinson) [reply README.md:34]: Great! Lmk if anything I said was confusing or if you need to talk anything through further.

Comment (haveanicedavid) [reply README.md]: @clevinson just updated, adding the makefile command `make install-local-ledger` which will clone and build regen ledger into a project subfolder to be used by the other make commands.

This should make setup easier, and avoid any potential conflicts with binaries. LMK how it looks

File: makefile [hunk 2]
```diff
@@ -23,137 +26,52 @@ ifeq ($(UNAME), Darwin) # MacOS
...
-	@echo "git clone https://github.com/cosmos/cosmos-sdk -> git checkout $(COSMOS_SDK_VERSION) -> make build -> mv build/simd $HOME/go/bin/"
-endif
-ifeq ($(UNAME), Darwin) # macOS
-	@echo "git clone https://github.com/cosmos/cosmos-sdk -> git checkout $(COSMOS_SDK_VERSION) -> make build -> echo $GOPATH/bin sudo mv build/simd /usr/local/go/bin/ -> ln -s /usr/local/go/bin/simd /usr/local/bin/simd"
-endif
+.PHONY: install-local-ledger
+install-local-ledger:
+	rm -rf local-ledger
+	git clone --depth 1 --branch $(LEDGER_BRANCH) https://github.com/regen-network/regen-ledger.git local-ledger/temp
+	cd local-ledger/temp && yarn install && make build
```
Comment (clevinson) [inline makefile:33]: I don't think we have a `yarn install` in the ledger repo
```suggestion
	cd local-ledger/temp && make build
```

File: makefile [hunk 3]
```diff
@@ -23,137 +26,52 @@ ifeq ($(UNAME), Darwin) # MacOS
...
+	$(LEDGER) start --home $(CHAIN_HOME) --log_level debug
 
-.PHONY: query-balance
-query-balance:
-	simd q bank balances $(ALICE) --chain-id $(CHAIN_ID) --home $(CHAIN_HOME)
-	simd q bank balances $(USER2) --chain-id $(CHAIN_ID) --home $(CHAIN_HOME)
+# .PHONY: query-balance
+# query-balance:
+# 	$(LEDGER) q bank balances $(GENESIS_ACCT_ADDR) --chain-id $(CHAIN_ID) --home $(CHAIN_HOME)
+# 	$(LEDGER) q bank balances $(USER_ADDR) --chain-id $(CHAIN_ID) --home $(CHAIN_HOME)
```
Comment (clevinson) [inline makefile:69]: Are these intended to be commented out?

**Review by clevinson — COMMENTED** [review]
Few small nits, generally looks good!

**Comment by haveanicedavid** [comment]
@clevinson made a follow up task to track: https://github.com/regen-network/groups-ui/issues/43

**Pull request merged**